automate rmd-to-md routine + update gcube + verbosely include dubicube

sannegovaert opened the pull request:
- fix #86 
- fix #87 
- update on b3doc updates (changes in parameters)
- verbosely include dubicube tutorials

wlangera reviewed (approved):
thanks, looks good

wlangera commented:
> @wlangera I now also added dubicube, could you review again please?
> 
> My code first didn't work, but after installing iNEXT, a dependency of b3gbi it worked as expected. Is there any way to avoid these hidden dependencies to run this script?

I guess this is related to **b3gbi** then and not to **dubicube**? Does this occur when you run a **dubicube** function or a **b3gbi** function?

sannegovaert commented:
> > @wlangera I now also added dubicube, could you review again please?
> > My code first didn't work, but after installing iNEXT, a dependency of b3gbi it worked as expected. Is there any way to avoid these hidden dependencies to run this script?
> 
> I guess this is related to **b3gbi** then and not to **dubicube**? Does this occur when you run a **dubicube** function or a **b3gbi** function?

Yes, it is related to b3gbi.

The workflow is as follows:
- Install all b3verse packages
- Each Rmd file is listed in [this yml](https://github.com/b-cubed-eu/documentation/blob/update-gcube/src/rmd_to_md/vignettes.yml) file
- Fore each Rmd file, I check which libraries are used, and add these to dependencies in the yml file
- Check if all these dependencies are installed
- Convert from Rmd to md

I now added iNEXT in the dependencies list. But I wonder if there is a way to avoid these kind of hidden dependency errors?

wlangera commented:
> > > @wlangera I now also added dubicube, could you review again please?
> > > My code first didn't work, but after installing iNEXT, a dependency of b3gbi it worked as expected. Is there any way to avoid these hidden dependencies to run this script?
> > 
> > 
> > I guess this is related to **b3gbi** then and not to **dubicube**? Does this occur when you run a **dubicube** function or a **b3gbi** function?
> 
> Yes, it is related to b3gbi.
> 
> The workflow is as follows:
> 
> * Install all b3verse packages
> * Each Rmd file is listed in [this yml](https://github.com/b-cubed-eu/documentation/blob/update-gcube/src/rmd_to_md/vignettes.yml) file
> * Fore each Rmd file, I check which libraries are used, and add these to dependencies in the yml file
> * Check if all these dependencies are installed
> * [Convert](https://github.com/b-cubed-eu/documentation/blob/update-gcube/src/rmd_to_md/rmd_to_md.Rmd) from Rmd to md
> 
> I now added iNEXT in the dependencies list. But I wonder if there is a way to avoid these kind of hidden dependency errors?

If I install **b3gbi** via the R universe repo, I get a warning indeed that installation of iNEXT fails:

```r
install.packages("b3gbi", repos = "https://b-cubed-eu.r-universe.dev")
```
```
Installing package into ‘C:/R/library’
(as ‘lib’ is unspecified)
Warning in install.packages :
  dependency ‘iNEXT’ is not availabletrying URL 'https://b-cubed-eu.r-universe.dev/bin/windows/contrib/4.5/sha256-933576f8a52e387e6e39d6d20634098c7b21f694f7bc0319ecf1407ce6050b9a'
Content type 'application/zip' length 2604232 bytes (2.5 MB)
downloaded 2.5 MB

package ‘b3gbi’ successfully unpacked and MD5 sums checked
```

Which is not the case if I install from GitHub:

```r
remotes::install_github("b-cubed-eu/b3gbi")
```
```
Using GitHub PAT from the git credential store.
Downloading GitHub repo b-cubed-eu/b3gbi@HEADThese packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                          
2: CRAN packages only                           
3: None                                         
4: mgrs         (0.2.4 -> 6ad14411c...) [GitHub]
5: rnaturale... (1.0.1 -> 1.1.0       ) [CRAN]  

Enter one or more numbers, or an empty line to skip updates: 
Installing 1 packages: iNEXT
Installing package into ‘C:/R/library’
(as ‘lib’ is unspecified)trying URL 'https://cloud.r-project.org/bin/windows/contrib/4.5/iNEXT_3.0.1.zip'
Content type 'application/zip' length 1535916 bytes (1.5 MB)
downloaded 1.5 MB

package ‘iNEXT’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\ward_langeraert\AppData\Local\Temp\RtmpWcFF5m\downloaded_packages
── R CMD build ─────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file 'C:\Users\ward_langeraert\AppData\Local\Temp\RtmpWcFF5m\remotes56403c9e4c09\b-cubed-eu-b3gbi-0d2b5d0/DESCRIPTION' ... 
─  preparing 'b3gbi':
✔  checking DESCRIPTION meta-information ... 
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building 'b3gbi_0.5.6.tar.gz' (357ms)
   
Installing package into ‘C:/R/library’
(as ‘lib’ is unspecified)* installing *source* package 'b3gbi' ...
** this is package 'b3gbi' version '0.5.6'
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (b3gbi)
```


I will look into this

wlangera reviewed (approved):
Can you have a look at my comment about the order on the side bar?

Otherwise, this can be merged

sannegovaert commented:
> Can you have a look at my comment about the order on the side bar?
> 
> Otherwise, this can be merged

I leave this to @peterdesmet. He also should have a look before merging.

wlangera commented:
> > > @wlangera I now also added dubicube, could you review again please?
> > > My code first didn't work, but after installing iNEXT, a dependency of b3gbi it worked as expected. Is there any way to avoid these hidden dependencies to run this script?
> > 
> > 
> > I guess this is related to **b3gbi** then and not to **dubicube**? Does this occur when you run a **dubicube** function or a **b3gbi** function?
> 
> Yes, it is related to b3gbi.
> 
> The workflow is as follows:
> 
> * Install all b3verse packages

The installation instructions were incomplete (see [here](https://github.com/b-cubed-eu/b-cubed-eu.r-universe.dev/issues/7)). I will update this asap.

peterdesmet reviewed (approved):
Excellent work @sannegovaert!! I have removed the GBIF_API tutorials for now, since they use a different technology and are out of scope
